### ISSUE TITLE
No results found for the location search.

### ISSUE THREAD

**Issue by dam62500** [bradcornford/Googlmapper#82]
Hi,

i use  Mapper->location('8 allée du Général Benoît 69673 Bron'); 
At first all was working well and without any reason, it has stopped working, I even tried single locations like Paris or London but it stopped working for no reason.
and i have this error : No results found for the location search.

**Comment by bradcornford** [comment]
Hi there,

Google may be rate limiting your IP. Under the hood, the location method requests a search against Google's geocode service. Try using that link, replacing it with your search, and check for the status "OVER_QUERY_LIMIT" or "REQUEST_DENIED" using:

```
http://maps.googleapis.com/maps/api/geocode/json?address=Sheffield&sensor=false
```

**Comment by dam62500** [comment]
that's worked before and every locations stopped work :/

**Comment by dam62500** [comment]
Maybe you're right about IP, how i can check ? with lat and lng, that's work well but i don't have that.

**Comment by bradcornford** [comment]
The status should state that your being limited. 

You could add a line in the method "searchLocation" in mapper.php:

```
dd ($request);
```

**Comment by dam62500** [comment]
thanks, you're right. 

"""
{\n
   "error_message" : "You have exceeded your daily request quota for this API. We recommend registering for a key at the Google Developers Console: https://console.developers.google.com/apis/credentials?project=_",\n
   "results" : [],\n
   "status" : "OVER_QUERY_LIMIT"\n
}\n
"""
i'll fix that tomorrow.

**Comment by bradcornford** [comment]
Yeah, I've noticed that the API key isn't being sent with the request either. Which won't be helping. I'll look into an update tomorrow. I'll also add some logic to check for the quota / denied states and throw a more meaningful exception.

**Comment by bradcornford** [comment]
I've now pushed some updates to the develop branch, if you want to give that a go. It should be more verbose with its exceptions under those differing response types.

**Comment by dam62500** [comment]
i added my API key server in your function searchLocation, for the file_get_contents and now it's fixed.

**Comment by bradcornford** [comment]
Did you not get chance to check the develop branch?

**Comment by dam62500** [comment]
Yes but i'm not familiar with the Google API, why not add in the configuration file the server api key with web api key?

**Comment by bradcornford** [comment]
The API key should work with both Google Maps API, and Google Geocode Service. As such, you can just update your configuration file with you API key.

**Comment by dam62500** [comment]
Ah i get it, my bad. thanks.

**Comment by bradcornford** [comment]
If you can let me know how you get on, ill get this fix merged to master and versioned.

**Comment by bradcornford** [comment]
I have now merged this into master under current version v2.16.0/v1.12.0.
